cpp/server/ct-server.cc

```diff
@@ -202,11 +202,15 @@ void SequenceEntries(TreeSigner<LoggedCertificate>* tree_signer,
 
   while (true) {
     if (election->IsMaster()) {
-      store->CleanupOldEntries();
-
-      // TODO(alcutter): Probably don't need to blow up here
-      CHECK_EQ(util::Status::OK, tree_signer->SequenceNewEntries())
-          << "Problem sequencing new entries";
+      util::Status status(store->CleanupOldEntries());
+      if (!status.ok()) {
+        LOG(WARNING) << "Problem cleaning up old entries: " << status;
+      } else {
+        status = tree_signer->SequenceNewEntries();
```

Comment:
I agree that this is the most likely thing to happen, but I wouldn't want some minor issue in the not-critical cleanup hold up sequencing either. Hitting etcd again straightaway seems like the lesser evil?

If it wasn't for the contention on the mapping key, these two things could be completely separate, so at least we can limit the damage of a possible bug or issue in the cleanup, is my theory.